samples/bluetooth/broadcast_audio_assistant/src/main.c

```diff
@@ -467,31 +590,38 @@ int main(void)
 
 		scan_for_broadcast_source();
 
-		/* FIX NEEDED: It should be valid to assign BT_BAP_BIS_SYNC_NO_PREF
-		 * to bis_sync, but currently (2024-01-30), the broadcast_audio_sink
-		 * sample seems to reject it (err=19) while other sinks don't.
-		 *
-		 * Also, if the source contains more than one stream (e.g. stereo),
-		 * some sinks have been observed to have issues. In this case,
-		 * set only one bit in bis_sync, e.g. subgroup.bis_sync = BIT(1).
-		 *
-		 * When PA sync and BASE is parsed (see note in the scan_recv_cb function),
-		 * the available bits can be used for proper selection.
-		 */
-		subgroup.bis_sync = BT_BAP_BIS_SYNC_NO_PREF;
+		printk("Scan stopped, attempting to PA sync to the broadcaster with id 0x%06X\n",
+			selected_broadcast_id);
+		err = pa_sync_create();
+		if (err != 0) {
+			printk("Could not create Broadcast PA sync: %d, resetting\n", err);
+			return -ETIMEDOUT;
+		}
+
+		printk("Waiting for PA synced\n");
+		err = k_sem_take(&sem_pa_synced, K_FOREVER);
+		if (err != 0) {
+			printk("sem_pa_synced timed out, resetting\n");
+			return -ETIMEDOUT;
+		}
```

Comment:
Does it make sense to return `-ETIMEDOUT` when using `K_FOREVER`? Those seem mutually exclusive